services/docextractor/documents.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package docextractor
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"code.sajari.com/docconv"
+)
+
+type documentExtractor struct{}
+
+func (de *documentExtractor) Match(filename string) bool {
+	supportedExtensions := map[string]bool{
+		"doc":   true,
+		"docx":  true,
+		"pptx":  true,
+		"odt":   true,
+		"html":  true,
+		"pages": true,
+		"rtf":   true,
+	}
+	extension := strings.TrimPrefix(path.Ext(filename), ".")
+	return supportedExtensions[extension]
+}
+
+func (de *documentExtractor) Extract(filename string, r io.Reader) (string, error) {
+	convertersByExtensions := map[string]func(io.Reader) (string, map[string]string, error){
```

Comment:
Same, and actually could this be combined on the same map? So you can check both existence and the related converter?